src/julia.expmap.in

```diff
@@ -35,13 +35,16 @@
     JLJIT*;
     llvmGetPassPluginInfo;
 
-    /* Make visible so that linker will merge duplicate definitions across DSO boundaries */
-    _ZN4llvm3Any6TypeId*;
-
     /* freebsd */
     environ;
     __progname;
 
   local:
     *;
 };
+
+JL_LLVM_15.0 {
```

Comment:
Can we get this from the Makefile instead? We have a preference for detection over declaration of versions, since it better supports upgrades and people who use system libraries. Something like this (probably could be made better and more reliable somehow though):
```
objdump -T libLLVM.so | grep _ZN4llvm3Any6TypeId | cut -f 2 | cut -d ' ' -f 3 | head -n 1
```
```
nm -D --with-symbol-versions libLLVM.so | grep _ZN4llvm3Any6TypeId | head -n 1 | sed -e 's/.*@//'
```

Reply:
`deps/llvm-ver.make` is the default version managed by deps, and is not the version of LLVM actually in use, so should not be used outside of that recipe

Reply:
It seems like `LLVM_SHARED_LIB_NAME` or `LLVM_SHARED_LINK_FLAG` are both exported and used externally - Is that a bug?

Reply:
I pushed a change with this approach, which works for me with `USE_BINARYBUILDER_LLVM=0` (and the usual build)

If there's a way to use a custom LLVM that can diverge from this, let me know and I'll take a different approach to fix it